sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ExtRefEditorService.java

```diff
@@ -488,9 +489,18 @@ public void updateIedNameBasedOnLnode(SCL scl) {
                                                                 && Objects.equals(tlNode.getLnInst(), tCompasFlow.getExtReflnInst())
                                                                 && Utils.lnClassEquals(tlNode.getLnClass(), tCompasFlow.getExtReflnClass())
                                                                 && Objects.equals(tlNode.getPrefix(), tCompasFlow.getExtRefprefix()))
+                                                        .filter(tlNode -> {
+                                                            TCompasICDHeader tCompasICDHeader = PrivateUtils.extractCompasPrivate(tlNode, TCompasICDHeader.class)
+                                                                    .orElseThrow(() -> new ScdException(("The substation LNode with following attributes : IedName:%s / LdInst:%s / LnClass:%s / LnInst:%s  " +
+                                                                            "does not contain the needed (COMPAS - ICDHeader) private")
+                                                                            .formatted(tlNode.getIedName(), tlNode.getLdInst(), tlNode.getLnClass().getFirst(), tlNode.getLnInst())));
+                                                            return Objects.equals(tCompasFlow.getFlowSourceIEDType(), tCompasICDHeader.getIEDType())
+                                                                    && Objects.equals(tCompasFlow.getFlowIEDSystemVersioninstance(), tCompasICDHeader.getIEDSystemVersioninstance())
+                                                                    && Objects.equals(tCompasFlow.getFlowSourceIEDredundancy(), tCompasICDHeader.getIEDredundancy());
+                                                        })
                                                         .map(TLNode::getIedName)
                                                         .filter(StringUtils::isNotBlank)
```

Comment:
According to XSD specification, `isNotBlank` is not logic validation for iedName since it had a default value.<br/>
![Capture d’écran du 2024-07-10 14-53-35](https://github.com/com-pas/compas-sct/assets/76203423/a325ff4a-38de-4494-9b9f-671b0aa8dff7)

We can keep using TLNode without map and no need for `checkOnlyOneIed` method creation.
```suggestion
.reduce((tlNode, tlNode2) -> {
sclReportItems.add(SclReportItem.error("", ("Several LNode@IedName ('%s', '%s') are found in the bay '%s' for the following compas-flow attributes :" +
                " @FlowSourceIEDType '%s' @FlowSourceIEDredundancy '%s' @FlowIEDSystemVersioninstance '%s'").
                formatted(tlNode.getIedName(), tlNode2.getIedName(), tBay.getName(), tCompasFlow.getFlowSourceIEDType(), tCompasFlow.getFlowSourceIEDredundancy(), tCompasFlow.getFlowIEDSystemVersioninstance())));
return tlNode;
})
```

Reply:
Indeed thank you for this improvement, I deleted the StringUtils:isNotBlank which appears to be useless but kept the private method "checkOnlyOneIed " which makes the code far more readable from my POV.